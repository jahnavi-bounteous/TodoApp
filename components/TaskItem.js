import React from "react";
import {View, Text, StyleSheet} from "react-native";
import { IconButton } from 'react-native-paper';
import moment from 'moment';

const TaskItem = ({ item, onDelete, onEdit, onToggleImportant }) => {
    const formattedDate = item.date ? moment(item.date).format('DD/MM/YYYY') : '';
    const formattedTime = item.time ? moment(item.time, 'HH:mm').format('hh:mm A') : '';
    return (
      <View
        style={styles.container}>
        <View style={{ flex: 1 }}>
          <Text style={styles.titleText}>
            {item.title}
          </Text>
          <View style={{ flexDirection: 'column', alignItems: 'flex-start' }}>
            <Text style={styles.datetimeText}>
              {formattedDate}
            </Text>
            <Text style={styles.datetimeText}>
              {formattedTime}
            </Text>
          </View>
        </View>
        <IconButton
          icon={item.isImportant ? 'heart' : 'heart-outline'}
          color="#fff"
          onPress={() => onToggleImportant(item.id)}
        />
        <IconButton
          icon="pencil"
          color="#fff"
          onPress={() => onEdit(item)}
        />
        <IconButton
          icon="trash-can"
          color="#fff"
          onPress={() => onDelete(item.id)}
        />
      </View>
    );
  };

  const styles = StyleSheet.create({
    container:{
      backgroundColor: '#1e90ff',
      borderRadius: 10,
      paddingHorizontal: 10,
      paddingVertical: 8,
      marginBottom: 12,
      marginHorizontal: 10,
      flexDirection: 'row',
      alignItems: 'center',
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 2 },
      shadowOpacity: 0.8,
      shadowRadius: 3,
    },
    titleText:{
      color: '#fff', 
      fontSize: 20, 
      fontWeight: '800'
    },
    datetimeText:{
      color: '#fff', 
      fontSize: 10, 
      fontWeight: '800'
    }
  })

export default TaskItem;