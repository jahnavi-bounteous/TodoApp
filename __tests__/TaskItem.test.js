import React from 'react';
import renderer from 'react-test-renderer';
import TaskItem from '../components/TaskItem';

describe('<TaskItem />', () => {
  const item = {
    id: 1,
    title: 'Sample Task',
    date: '2024-04-01',
    time: '13:30',
    isImportant: true,
  };

  const onDelete = jest.fn();
  const onEdit = jest.fn();
  const onToggleImportant = jest.fn();

  test('renders correctly', () => {
    const tree = renderer
      .create(
        <TaskItem
          item={item}
          onDelete={onDelete}
          onEdit={onEdit}
          onToggleImportant={onToggleImportant}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('calls onDelete when delete button is pressed', () => {
    const component = renderer.create(
      <TaskItem
        item={item}
        onDelete={onDelete}
        onEdit={onEdit}
        onToggleImportant={onToggleImportant}
      />
    );
    const deleteButton = component.root.findByProps({ icon: 'trash-can' });
    deleteButton.props.onPress();
    expect(onDelete).toHaveBeenCalledWith(1);
  });

  test('calls onEdit when edit button is pressed', () => {
    const component = renderer.create(
      <TaskItem
        item={item}
        onDelete={onDelete}
        onEdit={onEdit}
        onToggleImportant={onToggleImportant}
      />
    );
    const editButton = component.root.findByProps({ icon: 'pencil' });
    editButton.props.onPress();
    expect(onEdit).toHaveBeenCalledWith(item);
  });

//   test('calls onToggleImportant when important button is pressed', () => {
//     const component = renderer.create(
//       <TaskItem
//         item={item}
//         onDelete={onDelete}
//         onEdit={onEdit}
//         onToggleImportant={onToggleImportant}
//       />
//     );
//     const importantButton = component.root.findByProps({ icon: 'heart-outline' });
//     importantButton.props.onPress();
//     expect(onToggleImportant).toHaveBeenCalledWith(1);
//   });
});
