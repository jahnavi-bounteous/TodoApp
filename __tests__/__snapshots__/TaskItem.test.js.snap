// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskItem /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1e90ff",
      "borderRadius": 10,
      "flexDirection": "row",
      "marginBottom": 12,
      "marginHorizontal": 10,
      "paddingHorizontal": 10,
      "paddingVertical": 8,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "800",
        }
      }
    >
      Sample Task
    </Text>
    <View
      style={
        {
          "alignItems": "flex-start",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 10,
            "fontWeight": "800",
          }
        }
      >
        01/04/2024
      </Text>
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 10,
            "fontWeight": "800",
          }
        }
      >
        01:30 PM
      </Text>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        color="#fff"
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="icon-button"
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        color="#fff"
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="icon-button"
      >
        <Text />
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "height": 40,
        "margin": 6,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "width": 40,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 20,
          "borderWidth": 0,
          "elevation": 0,
          "flex": 1,
          "overflow": "hidden",
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        color="#fff"
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 20,
              },
            ],
          ]
        }
        testID="icon-button"
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
